content/posts/efficient-navigation-in-vim.md

```diff
@@ -0,0 +1,140 @@
+
+## Clunky Movements
+
+**TL;DR: Use for very specific use cases only. Consider `gg` and `G` for getting to know a file.**
+
+In this section we'll discuss what can be referred to as "clunky movements". They let the user navigate larger distances in the buffer while sacrificing precision. These commands are less useful for exactly that reason. Vim can be a very efficient text editor by letting the user think about what they want to change, jumping precisely there with very few keystrokes, entering insert mode and performing a change with scalpel-like precision and finally exiting insert mode as soon as it's done. The movement commands in this section however, get the cursor around the document as a whole while it's hard for the user to predict at a glance, which line and column exactly they will land on.
+
+A good example are the `H`, `M` and `L` keys which take the cursor to the top, the middle or the bottom of the current view port respectively. While this is a very big movement with just one keystroke, it's highly likely that they won't exactly get the cursor to the line needed but rather will have to be accompanied by pressing `j` or `k` multiple times which will result in much more thinking and many more keystrokes than what can be achieved with other methods. Similar are the `{` and `}` keys which take the cursor to the next paragraph (a block of text delimited by blank lines).
+
+A useful exception to that are the `gg` and `G` commands which take the cursor to the first or the last line of a buffer respectively. It's easy and effortless to predict where these movements will take. Especially the `G` command can be quite useful because it allows to append to a file with just two keystrokes (`G` followed by `o`).
+
+## Plugins
+
+**TL;DR: Install only the necessary plugins. Check out fzf!**
+
+So far, we've only talked about features that are built into Vim or that can be configured on a vanilla installation of Vim. However, there are many useful plugins to can make navigating Vim even more efficient. It is important to carefully pick them though as any plugin can make Vim slower and/or less stable.
+
+### fzf
+
+One of the most useful plugins is [fzf](https://github.com/junegunn/fzf.vim) which is a great one to have in general. It offers many helpful commands like `:BLines` for searching the current buffer with intelligent fuzzy matching or `:Rg` for even searching the whole project for specific patterns which, in contrast to the other methods we've looked at so far, lets the user navigate between files. fzf is a tool that can be used for jumping between files and buffers but also between different locations within them. It is highly recommended to check it out and add the following lines to the configuration:
```

Comment:
`It is highly recommended to check it out` sounds a bit detached. How about `I highly recommend that you check it out` or at least `It is highly recommended that you check it out`